R/mostRelSys.R

```diff
@@ -0,0 +1,93 @@
+#mostRelSys.R
+#
+#purpose: find the most related system with the given gene name
+#Version:1.0.0
+#Data:2019.3
+#Author:Liwen Zhuang
+#
+#Input:System data, normalized STRING edge score, and gene name of interest
+#Output:The most related system in the given system data
+#Dependencies:None
+#
+#
+#===PARAMETERS ====
+#db: database which stores the given system data
+#geneName: HGNC symbol of the interested gene
+#
+#
+#===FUNCTIONS =====
+#
+#' mostRelSys
+#'
+#' main funcion, output the most related system of a given gene.
+#'
+#' @param db dataframe,database which stores the system data.
+#' @param geneName string,the interested HGNC symbol
+#' @return A name of the System which is most related to the given gene name.
+#' @export
+#' @example
+#'
+#' myURL <- paste0("http://steipe.biochemistry.utoronto.ca/abc/assets/","STRINGedges-2019-03-14.RData")
+#' load(url(myURL))
+#'
+#' myDB <- fetchData("SysDB")
+#' mostRelSys(myDB,"CSK")
+#'
+mostRelSys <- function(db,geneName){
+  numSys<- length(names(SyDBgetRootSysIDs(db)))
+  scoreindex<-1
+  scorelist<-c()
+  for (sysName in names(SyDBgetRootSysIDs(db))){
+    totalscore<-0 #total score of edges between input and element in this system
+    totaledges<-0 #total number of edges existed
+    SymbolList<-unlist(SyDBgetSysSymbols(db, sysName))#vectorize the list
+    for (SysSymbols in SymbolList) {
+      if (fetchScore(SysSymbols,geneName) != -1){#edge exist in normalized STRING database
+        totalscore<-totalscore+ fetchScore(SysSymbols,geneName)
+        totaledges<-totaledges+1
+      }
+    }
+    finalScore <- totalscore/length(SymbolList)
+    scorelist[scoreindex]<-finalScore
+    scoreindex<-scoreindex+1
+    cat("total interaction found in system ",sysName,"is ",totaledges,"with final score of ",finalScore,".\n")
+  }
+  return(names(SyDBgetRootSysIDs(db))[which.max(scorelist)])
```

Comment:
Probably better as a named vector, or a two column data frame with symbols and scores. Sorted.